src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py

```diff
@@ -11,13 +11,215 @@
 """
 
 import os
+from typing import Optional
 
 from zowe.core_for_zowe_sdk import SdkApi
 from zowe.core_for_zowe_sdk.exceptions import FileNotFound
 from zowe.zos_files_for_zowe_sdk.constants import FileType, zos_file_constants
 
 _ZOWE_FILES_DEFAULT_ENCODING = zos_file_constants["ZoweFilesDefaultEncoding"]
 
+
+class DatasetOption:
+
+    def __init__(
+        self,
+        like: Optional[str] = None,
+        volser: Optional[str] = None,
+        unit: Optional[str] = None,
+        dsorg: Optional[str] = None,
+        alcunit: Optional[str] = None,
+        primary: Optional[int] = None,
+        secondary: Optional[int] = None,
+        dirblk: Optional[int] = None,
+        avgblk: Optional[int] = None,
+        recfm: Optional[str] = None,
+        blksize: Optional[int] = None,
+        lrecl: Optional[int] = None,
+        storclass: Optional[str] = None,
+        mgntclass: Optional[str] = None,
+        dataclass: Optional[str] = None,
+        dsntype: Optional[str] = None,
+    ) -> None:
+        self.__like = like
+        self.volser = volser
+        self.unit = unit
+        self.dsorg = dsorg
+        self.alcunit = alcunit
+        self.primary = primary
+        self.secondary = secondary
+        self.dirblk = dirblk
+        self.avgblk = avgblk
+        self.recfm = recfm
+        self.blksize = blksize
+        self.lrecl = lrecl
+        self.storclass = storclass
+        self.mgntclass = mgntclass
+        self.dataclass = dataclass
+        self.dsntype = dsntype
+
+    @property
+    def volser(self) -> Optional[str]:
+        return self.__volser
+
+    @volser.setter
+    def volser(self, volser: Optional[str]):
+        self.__volser = volser
+
+    @property
+    def unit(self) -> Optional[str]:
+        return self.__unit
+
+    @unit.setter
+    def unit(self, unit: Optional[str]):
+        self.__unit = unit
+
+    @property
+    def dsorg(self) -> Optional[str]:
+        return self.__dsorg
+
+    @dsorg.setter
+    def dsorg(self, dsorg: Optional[str]):
+        if dsorg not in ("PO", "PS", None):
+            raise ValueError("'dsorg' must be 'PO', 'PS', or None")
+        self.__dsorg = dsorg
+
+    @property
+    def alcunit(self) -> Optional[str]:
+        return self.__alcunit
+
+    @alcunit.setter
+    def alcunit(self, alcunit: Optional[str]):
+        if alcunit is None:
+            if self.like is not None:
+                self.alcunit = None
```

Comment:
Should this be self.__alcunit? I run into an issue with here when I only specify a `like` data set.

Reply:
Looks good, thanks